lua/which-key/keys.lua

```diff
@@ -211,7 +211,14 @@ function M.parse_mappings(mappings, value, prefix)
       end
       if mapping.cmd and type(mapping.cmd) == "function" then
         table.insert(M.functions, mapping.cmd)
-        mapping.cmd = string.format([[<cmd>lua require("which-key").execute(%d)<cr>]], #M.functions)
+        if mapping.opts.expr then
+          if _G.which_key_execute == nil then
+            _G.which_key_execute = require("which-key").execute
+          end
+          mapping.cmd = string.format([[v:lua.which_key_execute(%d)]], #M.functions)
```

Comment:
The `_G` is not needed here, you can change the cmd to something like:
```lua
string.format([[luaeval('require("which-key").execute(%d)')]], #M.functions)
```

Reply:
OK